### PR TITLE
Add challenge friend

### PR DESCRIPTION
**Pull request by vikdoro**
The button had to be lowercase and for other design reasons I didn't use the kwc-button. It's in my to-do ilst to work on the kwc-button to make it a bit more styleable with some css mixins and also display and dimension properties on its shadow host.

### CODE REVIEW

**Review by russormes — CHANGES_REQUESTED** [review]
One observation is that there is no way to not display the challenge a friend button. Could we add a property to the component to allow this to be configured by its parent component?

**Comment by vikdoro** [comment]
@russormes I updated the pr with a property to optionally hide the cta. Thanks!

File: kwc-challenge-set.html [hunk 1]
```diff
@@ -371,6 +420,18 @@ <h3 class="title-title">
...
+                },
+                buttonLabel: {
+                    type: String,
+                    computed: '_computeButtonLabel(_completed, progress)'
+                },
+                /**
+                 * Flags if challenge-friend button is hidden
+                 * @type {Boolean}
+                 */
+                _hideChallengeFriend: {
```
Comment (murilopolese) [inline kwc-challenge-set.html]: I guess if this is something we'll be setting from "outside" this should not be prefixed with a `_`, no?

Comment (vikdoro) [reply kwc-challenge-set.html]: Totes :) Thanks!

**Pull request merged**